resources/fetchers/ecr_fetcher.go

```diff
@@ -56,7 +56,7 @@ type ECRFetcherConfig struct {
 	KubeConfig                    string `config:"Kubeconfig"`
 }
 
-type EcrRepository ecr.Repository
+type EcrRepository types.Repository
```

Comment:
To completely finish the amazing segregation that you made, can we replace the `types. Repository` with a type of our own? 

Same as you did in the IAM and the EKS fetchers

That way, only the AWS provider will be aware of the AWS interface